combustion-android-ble/src/main/java/inc/combustion/framework/ble/NetworkManager.kt

```diff
@@ -72,6 +73,76 @@ internal class NetworkManager(
         var mutableDeviceProximityFlow: MutableSharedFlow<DeviceDiscoveredEvent>,
     )
 
+    inner private class NodeDeviceManager(
+        val owner: LifecycleOwner
+    ) {
+        private val connectedNodes : MutableMap<String, NodeBleDevice> = mutableMapOf()
+        private val ignoredNodes : MutableSet<String> = mutableSetOf()
+
+        fun sendNodeRequest(deviceId: String, request: GenericNodeRequest, completionHandler: (Boolean, Any?) -> Unit) {
+            connectedNodes[deviceId]?.sendNodeRequest(request) { status, data ->
+                completionHandler(status, data)
+            } ?: run {
+                completionHandler(false, null)
+            }
+        }
+
+        // TODO: this needs to support removing the devices as well as adding them. Perhaps that should be another flow
+        fun subscribeToNodeFlow(probeManager: ProbeManager) {
+            owner.lifecycleScope.launch(
+                CoroutineName("NodeConnectionFlow")
+            ) {
+                probeManager.nodeConnectionFlow
+                    .collect { deviceIds ->
+                        deviceIds.forEach { deviceId ->
+                            when (val node = devices[deviceId]) {
+                                is DeviceHolder.ProbeHolder -> NOT_IMPLEMENTED("Unsupported device type")
+                                is DeviceHolder.RepeaterHolder -> {
+                                    updateConnectedNodes(node.repeater)
```

Comment:
Checking how this works.  Does this (re)request feature flags and serial numbers to all connected nodes whenever the connection changes for any node?

Reply:
The intent was for that to only happen once for each one but it looks like with some refactoring I did I dropped one check so the serial number message might get sent a lot. I'll update that. 

The `ignoredNodes` and `connectedNodes` maps/sets are supposed to track all the nodes that have been seen so far and only send messages for those it doesn't know about.

Reply:
I think I've taken care of the extra serial number messages.